docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -12,3 +12,225 @@ pageClass: migration-guide
+  - [`AutoColumnSize`](@/api/autocolumnsize.md)
+  - [`AutoRowSize`](@/api/autorowsize.md)
+  - `TouchScroll`
+  - [`UndoRedo`](@/api/undoredo.md)
+:::
+
+#### Migrating to Handsontable 12.0
+
+If you want your [custom plugin](@/guides/building-and-testing/plugins.md) to still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call, set your plugin's [`SETTING_KEYS`](@/api/baseplugin.md#setting-keys) to `true`:
+
+```js
+static get SETTING_KEYS() {
+  return true;
+}
+```
+
+However, in most cases, it's better to provide an explicit list of configuration options that your custom plugin observes. For details, see the [Plugins](@/guides/building-and-testing/plugins.md) guide.
+
+## Step 3: Adjust to the `afterDocumentKeyDown` changes
+
+Handsontable 12.0.0 changes how the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook works.
+
+#### Before
+
+When you used a keyboard to navigate an internal Handsontable instance (e.g., a context menu), the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook got fired on every key press.
+
+```js
+afterDocumentKeyDown() {
+  // when you navigate the grid, the console message gets logged on every key press
+  // when you navigate the context menu, the console message also gets logged on every key press
+  console.log('key pressed');
+}
+```
+
+#### After
+
+When you use a keyboard to navigate some internal instances of Handsontable (e.g., a context menu) hook may not get fired anymore.
+
+```js
+afterDocumentKeyDown() {
+  // when you navigate the grid, the console message gets logged on every key press
+  // when you navigate the context menu, the console message doesn't get logged at all
+  console.log('key pressed');
+}
+```
+
+This change may affect the following areas of Handsontable:
+- [Context menu](@/api/contextmenu.md)
+- [Dropdown menu](@/api/dropdownmenu.md)
+
+#### Migrating to Handsontable 12.0
+
+You can't change this behavior by using any of Handsontable's APIs.
+
+## Step 4: Adjust to the default keyboard shortcuts changes
+
+::: tip
+These changes don't affect your custom keyboard shortcuts.
+:::
+
+Handsontable 12.0 introduces a new keyboard shortcuts API, [`ShortcutManager`](@/api/shortcutmanager.md).
+
+[`ShortcutManager`](@/api/shortcutmanager.md) lets you easily manage your custom keyboard shortcuts,
+but also changes how Handsontable defines its default keyboard shortcuts.
+Now, nearly all default keyboard shortcuts are defined explicitly.
+
+We took this opportunity to improve the behavior of Handsontable's default keyboard shortcuts, so they:
+- Better match their counterparts in popular spreadsheet software
+- Improve the consistency of keyboard navigation
+- Create a more intuitive user experience
+
+#### The Meta key (<kbd>Ctrl</kbd> vs <kbd>Cmd</kbd>)
+
+Handsontable 12.0 splits the cross-platform Meta key (<kbd>Ctrl</kbd>/<kbd>Cmd</kbd>) into two separate keys.
```

Comment:
This and the following information was reworded in the blog post. Please update here.

Reply:
Done: [4588986](https://github.com/handsontable/handsontable/pull/9271/commits/4588986124f5baddf163c2da25cb5ec2a41b53ca)